### PR TITLE
[Feature] Support session reset system message

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

Support displaying a system for when the another user resets the session with you.

### Dependencies

Needs releases with:

- [ ] https://github.com/wireapp/wire-ios-data-model/pull/1088

### CODE REVIEW

File: Wire-iOS Tests/ConversationMessageCell/ConversationSystemMessageTests.swift [hunk 1]
```diff
@@ -70,6 +70,18 @@ final class ConversationSystemMessageTests: ConversationCellSnapshotTestCase {
 
         verify(message: message)
     }
+    
+    func testSessionReset() {
+        let message = MockMessageFactory.systemMessage(with: .sessionReset, users: 1, clients: 1, sender: otherUser)!
+
+        verify(message: message)
```
Comment (billypchan) [inline Wire-iOS Tests/ConversationMessageCell/ConversationSystemMessageTests.swift:77]: We may not need to snapshot all device widths for these tests, but we can clean them up later.

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 2]
```diff
@@ -533,6 +533,10 @@
 "content.system.missing_messages.title" = "You haven’t used this device for a while. Some messages may not appear here.";
 "content.system.missing_messages.subtitle_start" = "Meanwhile,";
 
+"content.system.session_reset" = "%@ was unable to decrypt some of your messages but has solved the issue.";
+"content.system.session_reset.self" = "You was unable to decrypt some of your messages but has solved the issue.";
+"content.system.session_reset.affected" = "This affected all conversation you share together.";
```
Comment (billypchan) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings:538]:
```suggestion
"content.system.session_reset.affected" = "This affected all conversations you share together.";
```

Comment (billypchan) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings:538]: I think we can have a simpler sentence.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift [hunk 3]
```diff
@@ -707,6 +711,50 @@ class ConversationIgnoredDeviceSystemMessageCellDescription: ConversationMessage
     
 }
 
+class ConversationSessionResetSystemMessageCellDescription: ConversationMessageCellDescription {
+    typealias View = ConversationSystemMessageCell
+    
+    var message: ZMConversationMessage?
+    var delegate: ConversationMessageCellDelegate?
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:718]: weak?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift [hunk 4]
```diff
@@ -707,6 +711,50 @@ class ConversationIgnoredDeviceSystemMessageCellDescription: ConversationMessage
...
+    var actionController: ConversationMessageActionController?
+    
+    var topMargin: Float = 0
+    var isFullWidth: Bool = true
+    var supportsActions: Bool = false
+    var showEphemeralTimer: Bool = false
+    var containsHighlightableContent: Bool = false
+    var accessibilityIdentifier: String? = nil
+    var accessibilityLabel: String? = nil
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:727]:
```suggestion
    var accessibilityIdentifier: String?
    var accessibilityLabel: String?
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift [hunk 5]
```diff
@@ -707,6 +711,50 @@ class ConversationIgnoredDeviceSystemMessageCellDescription: ConversationMessage
...
+        return part1 + " " + part2
+    }
+    
+    static func localizedMessagePart1(_ sender: UserType) -> String {
+        if sender.isSelfUser {
+            return "content.system.session_reset.self".localized
+        } else {
+            return "content.system.session_reset".localized(args: sender.name ?? "")
+        }
+    }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:751]: How about make this as a method of UserType? e.g. 
```
private extension UserType {
  var localizedMessagePart1: String {
  ...
  }
}
```

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift [hunk 6]
```diff
@@ -707,6 +711,50 @@ class ConversationIgnoredDeviceSystemMessageCellDescription: ConversationMessage
...
+    
+    static func localizedMessagePart2() -> String {
+        return "content.system.session_reset.affected".localized
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:754]: maybe we do not need a static method if only called once?

**Review by billypchan — APPROVED** [review]

**Review by billypchan — COMMENTED** [review]
minor issues, but ok to merge

File: Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 2 above)
Comment (typfel) [reply Wire-iOS/Resources/Base.lproj/Localizable.strings:538]: I'm sure the copy will be improved but I'm just putting was given to me.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift (hunk 5 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:751]: I'm just following the way all other system messages are implemented. I'm sure we can improve them but would do that in a separate PR.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift (hunk 6 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:754]: just following the conventions of how system messages are implemented.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift (hunk 3 above)
Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationSystemMessageCell.swift:718]: Yes, I'll check if we need it.

**Review by billypchan — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Comment by typfel** [comment]
@billypchan Why did you merge this? This should have gone into my feature branch

**Pull request merged**